src/routes/health-check/validation.ts

```diff
@@ -0,0 +1,39 @@
+import type { Context } from 'hono';
+import { z } from 'zod';
+import type { ZodSchema } from 'zod';
+type EnvKeys = keyof EnvVars;
+
+// NOTE: please update the schema of new environment variables here
+const envSchema = z.object(
+	{
+		BASE_URL: z
+			.string({ message: 'required for the server to run' })
+			.url({ message: 'must be a valid URL' }),
+		SENDER_EMAIL: z
+			.string({ message: 'required for emails to be sent' })
+			.email({ message: 'must be a valid email address' }),
+		SENDGRID_API_KEY: z
+			.string({ message: 'required for emails to be sent' })
+			.min(1, { message: 'required for emails to be sent' }),
+		NODE_ENV: z
+			.enum(['development', 'local', 'production'], { message: 'must be one of the following: development, local, production' })
+	} satisfies Record<EnvKeys, ZodSchema>
+);
+
+function checkEnvVars(context: Context<EnvironmentBindings>) {
+	// NOTE: please add new environment variables here
+	const env: EnvVars = {
+		BASE_URL: context.env.BASE_URL,
+		SENDER_EMAIL: context.env.SENDER_EMAIL,
+		SENDGRID_API_KEY: context.env.SENDGRID_API_KEY,
+		NODE_ENV: context.env.NODE_ENV
+	};
```

Comment:
Tied to the previous comment: if we pass only `env` to this function, we don't need to reconstruct the env object here.

Reply:
Oooh yeah that simplifies the code and means the developer only needs to update the schema thanks 👍